src/meta/src/hummock/hummock_manager.rs

```diff
@@ -1559,4 +1558,34 @@ where
         }
         Ok((deleted, remain))
     }
+
+    /// Extends `ssts_to_delete` according to object store full scan result.
+    /// Caller should ensure `sst_ids` doesn't include any SSTs belong to a on-going version write.
+    /// That's to say, these sst_ids won't be seen in either `commit_epoch` or
+    /// `report_compact_task`.
+    #[named]
+    pub async fn extend_ssts_to_delete_from_scan(&self, sst_ids: &[HummockSstableId]) {
+        let tracked_sst_ids: HashSet<HummockSstableId> = HashSet::from_iter({
+            let versioning_guard = read_lock!(self, versioning).await;
+            let mut tracked_sst_ids = versioning_guard.current_version.get_sst_ids();
```

Comment:
I think we should start from `checkpoint_version` instead of `current_version`?

Reply:
Also, shall we use a HashSet to store `tracked_sst_ids`?

Reply:
> I think we should start from `checkpoint_version` instead of `current_version`?

Thanks for spotting this fatal one.
Fixed by getting tracked SST ids from current_version + deltas' removed_sst_ids.